protocol/README.md

```diff
@@ -0,0 +1,405 @@
+# Astoria Protocol
+
+*Version 1.0.0*
+
+This document defines the protocol used for Astoria and it's various
+services. This document should be used when implementing clients for
+Astoria.
+
+This document uses semantic versioning and should have changes to it
+reviewed and approved appropriately to ensure that any breaking changes
+are dealt with appropriately.
+
+Where appropriate, schemas are provided alongside this document and must
+be kept up to date as the protocol develops over time.
+
+## Protocol Overview
+
+The Astoria is an IPC mechanism between multiple independent programs
+that hold individual state on a robot.
+
+There are two categories of program that interact using the Astoria
+Protocol:
+
+-   Astoria Services  
+    -   A service is a program that holds and manages its own state.
+    -   The state usually reflects a *domain* that the service is
+        controlling, e.g running usercode.
+    -   Each service must be defined in the protocol specification.
+    -   Additionally, each service has a *slug*, usually the name of the
+        service.
+
+-   Astoria Clients  
+    -   A client is a program that interacts with Astoria but does not
+        expose state to other programs using Astoria
+    -   The client is able to read the state of services but may not
+        necessarily be able to manipulate the state.
+    -   A client is usually associated with a user, such as a web
+        interface or command line client.
+
+Services and clients communicate using defined message types over a
+transport, defined later in this specification.
+
+## Transport
+
+The protocol sends and receives messages using MQTT v5. Other versions
```

Comment:
> Other versions of MQTT are not supported

Question: why?

Reply:
I've realised we've specified exactly v5, updated to v5 or later. [9d1bf66](https://github.com/srobo/astoria/pull/204/commits/9d1bf6690f1dde6b6b332bb8480c5680f91b2676)